morph/cli.py

```diff
@@ -116,6 +131,109 @@ def cmd_count(args):
         print '%d\t%s' % (c, m.show().encode('utf-8'))
 
 
+def cmd_next(args):
+    notes_path = args.notes
+    prio_path = args.prio
+    mizer = MIZERS[args.mizer]
+
+    known = load_db('known')
+
+    candidates = defaultdict(list)
+    with codecs.open(notes_path, 'r', 'utf-8') as f:
+        for line in f.readlines():
+            note = line.strip()
+            text = note.split('\t', 1)[0]
+            unknowns = [m for m in mizer.getMorphemesFromExpr(text)
+                        if m not in known.db]
+            if len(unknowns) == 1:
+                candidates[unknowns[0].show()].append(note)
```

Comment:
This is a "do I understand this right?"-comment with some brainstorming at the end...

If the format of the "note file" is a text file with sentences (like a novel or a subtitle without timing/styling information), the split(tab) is out of place, because a tab shouldn't be treated differently.
If the format is a text file with "MorphMan-style" lines and fields like you explained before, I'm confused by the list comprehension (suggesting there is more than one morpheme in the first field) and that `unknown[0].show()` would expand to `note`, so the whole point of using a candidates dictionary is undermined (inserting x for the key x).

So I think I understand you use a Anki-note like format, where the fields are tab separated and the first field represents "the front"/Expression of the card. I took me some time to come to this conclusion because
 - `file of newline-terminated notes, each tab-separated fields starting with the text` was ambiguous for me (I read it like the fields start with text)
 - the name `prio` suggests a priority, not a filter and thus the meaning of this command got unclear for me

Because this function has two atomic parts, I also came up with some possibly useful other commands that can create this command. Something like:
 - `mm filter` where the first argument is a file containing notes, and the second argument a file with a MorphMan-style list of morphemes (something like grep with multiple morphmemes)
 - `mm filter-vocab-card`, `mm filter-comprehension-card`, `mm filter-fresh-card` which act on a file with note-like contents; the names are corresponding to the tags, so `mm filter-vocab-card` does what this currently loop does
   - alternatively something like a `mm filter-num-unknown --equal 1` is thinkable (and something like `mm filter-num-unmatures --greater 1 --less 3`)
 - `mm rank` which takes a note file and adds a new first field corresponding to it's MorphMan index, sharing the functionality from `updateNotes()` (so sorting from command line is possible)

If you don't want to implement these functions (now or ever), I will still accept this pull request.